x/stakeibc/keeper/icacallbacks_undelegate_test.go

```diff
@@ -287,86 +292,153 @@ func (s *KeeperTestSuite) TestGetLatestCompletionTime_Success() {
+	hostZone := stakeibc.HostZone{
+		ChainId: chainId,
+	}
+	callbackArgs := types.UndelegateCallback{
+		EpochUnbondingRecordIds: []uint64{1},
+	}
+	completionTime := s.Ctx().BlockTime()
+	_, err := s.App.StakeibcKeeper.UpdateHostZoneUnbondings(s.Ctx(), completionTime, hostZone, callbackArgs)
+	s.Require().EqualError(err, "Unable to find epoch unbonding record for epoch: 1: key not found")
+}
+
 // Test failure case - HostZoneUnbonding DNE
+func (s *KeeperTestSuite) TestUpdateHostZoneUnbondings_HostZoneUnbondingDNE() {
+	hostZone := stakeibc.HostZone{
+		ChainId: chainId,
+	}
+	epochUnbondingRecord := recordtypes.EpochUnbondingRecord{
+		EpochNumber: 1,
+		// No hzu!
+		HostZoneUnbondings: []*recordtypes.HostZoneUnbonding{},
+	}
+	s.App.RecordsKeeper.SetEpochUnbondingRecord(s.Ctx(), epochUnbondingRecord)
+	callbackArgs := types.UndelegateCallback{
+		EpochUnbondingRecordIds: []uint64{1},
+	}
+	completionTime := s.Ctx().BlockTime()
+	_, err := s.App.StakeibcKeeper.UpdateHostZoneUnbondings(s.Ctx(), completionTime, hostZone, callbackArgs)
+	s.Require().EqualError(err, "Host zone not found (GAIA) in epoch unbonding record: 1: key not found")
+}
+
 // Test failure case - Amount too big
+func (s *KeeperTestSuite) TestUpdateHostZoneUnbondings_AmountTooBig() {
+	hostZone := stakeibc.HostZone{
+		ChainId: chainId,
+	}
+	// Set up EpochUnbondingRecord, HostZoneUnbonding and token state
+	hostZoneUnbonding := recordtypes.HostZoneUnbonding{
+		HostZoneId:    chainId,
+		Status:        recordtypes.HostZoneUnbonding_BONDED,
+		StTokenAmount: math.MaxUint64,
+	}
+	// Create two epoch unbonding records (status BONDED, completion time unset)
+	epochUnbondingRecord := recordtypes.EpochUnbondingRecord{
+		EpochNumber:        1,
+		HostZoneUnbondings: []*recordtypes.HostZoneUnbonding{&hostZoneUnbonding, &hostZoneUnbonding},
+	}
+	s.App.RecordsKeeper.SetEpochUnbondingRecord(s.Ctx(), epochUnbondingRecord)
+	callbackArgs := types.UndelegateCallback{
+		EpochUnbondingRecordIds: []uint64{1, 2},
+	}
+	completionTime := time.Now().Add(time.Second * time.Duration(10))
+	_, err := s.App.StakeibcKeeper.UpdateHostZoneUnbondings(s.Ctx(), completionTime, hostZone, callbackArgs)
+	s.Require().EqualError(err, "Could not convert stTokenAmount to int64 in redeem stake | overflow: unable to cast 18446744073709551615 of type uint64 to int64: unable to cast to safe cast int")
+}
 
 // BurnTokens Tests
 func (s *KeeperTestSuite) TestBurnTokens_Success() {
 	_ = s.SetupUndelegateCallback()
 
 	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), chainId)
-	s.Require().True(found)
+	s.Require().True(found, "host zone found")
 
 	zoneAccount, err := sdk.AccAddressFromBech32(hostZone.Address)
-	s.Require().NoError(err)
+	s.Require().NoError(err, "zoneAccount is valid")
 	s.Require().Equal(s.App.BankKeeper.GetBalance(s.Ctx(), zoneAccount, stAtom).Amount.Int64(), int64(300_010), "initial token balance is 300_010")
```

Comment:
flop expected and actual args in these burn tests. And grab initial balance from test case struct (`tc.initialState.zoneAccountBalance`)

Reply:
good catch, done